### DISCUSSION TITLE
Is ser2net reliable?

### DISCUSSION THREAD

**Discussion by eigenphase** [Koenkk/zigbee2mqtt#22539] [General]
Anyone using this and wanna share experience?

I know it should explicitly not be used on WiFi. For me, it would be wired Ethernet (RPi0 via PoE hat). But also on wired Ethernet hiccups can happen. Trivially switch or router lags/reboots, cable plug/unplug etc. Of course, in such situations there will be a brief drop but how does it recover? Would it need to wait for a TCP timeout for example?

I’m currently running z2m on my RPi0 directly but I feel it’s not that solid. Starting takes minutes and load is high. Looking for ways to improve and wonder if putting z2m on my HA box and ser2net is a better solution

**Comment by hellcry37** [comment]
Excellent reliability I've used this solution for at least 1y no issues at all. (LAN connection)

**Reply by eigenphase** [reply]
Thank you 😊

**Reply by eigenphase** [reply]
Ok I just set it up and give it a go!

Just out of curiosity. It seems there is socat as well. Is ser2net or socat the more reliable solution?

**Reply by francisp2** [reply]
https://www.lucid-control.com/accessing-usb-io-module-network-linux-socat
